google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorkerPool.java

```diff
@@ -421,4 +437,23 @@ FlowController.LimitExceededBehavior limitExceededBehavior() {
   BigQueryWriteClient bigQueryWriteClient() {
     return client;
   }
+
+  // Callback used for populating updated schema from response.
+  private class ComputeUpdatedSchemaCallback implements ApiFutureCallback<AppendRowsResponse> {
+    private final String streamName;
+
+    ComputeUpdatedSchemaCallback(String streamName) {
+      this.streamName = streamName;
+    }
+
+    @Override
+    public void onFailure(Throwable t) {
+      // No Op during failure.
+    }
+
+    @Override
+    public void onSuccess(AppendRowsResponse response) {
+      streamNameToUpdatedSchema.put(streamName, response.getUpdatedSchema());
```

Comment:
After you refreshWriter, you need to mark the entry here as null. I think it is better we keep a schema on the Writer level.

Reply:
The problem is there could be multiple stream writers that use the same stream name that all need refreshWriter to be triggered whenever there is a updated schema

So we can't directly nullify the updated schema for a given stream name, otherwise some streamwriter might not be able to get the updated schema correctly.

Reply:
As discussed offline, changed to use timestamp pattern 

If the timestamp used in the current stream writer is older than the updated schema version,change to use updated schema